x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalServiceTests.java

```diff
@@ -1291,6 +1292,9 @@ public void testChunkingLargeDocument() throws InterruptedException {
             numResponsesPerBatch[i] = ElasticsearchInternalService.EMBEDDING_MAX_BATCH_SIZE;
         }
         numResponsesPerBatch[numBatches - 1] = numChunks % ElasticsearchInternalService.EMBEDDING_MAX_BATCH_SIZE;
+        if (numResponsesPerBatch[numBatches - 1] == 0) {
+            numResponsesPerBatch[numBatches - 1] = ElasticsearchInternalService.EMBEDDING_MAX_BATCH_SIZE;
```

Comment:
Can you clarify why we need to do this?

Reply:
Dave wrote this code. Anyway:

- let's call `ElasticsearchInternalService.EMBEDDING_MAX_BATCH_SIZE)` S for now
- `numBatches` (`B` from now on) is chosen between 3 and 6 
- `numChunks` (`C` from now on) is chosen such that it needs `B` batches,
   i.e. between `(B-1)*S+1` (so that the last batch has size 1), and `B*S` (so that the last batch has size S)
- batches `0...B-2` have size `S` (they're full)
- the last batch `B-1` has size between `1` and `S`; most of the times this is `C%S` (the remainder after division), except when it's full: then the remainder is `0`, but the batch size is `S`

Does that help?

Reply:
I've simplified Dave's logic for this test. Hope it's clearer now.